src/Fixer/FunctionNotation/VoidReturnFixer.php

```diff
@@ -0,0 +1,254 @@
+            [
+                new VersionSpecificCodeSample(
+                    "<?php\nfunction foo(\$a) {};",
+                    new VersionSpecification(70100)
+                ),
+            ],
+            'Rule is applied only in a PHP 7.1+ environment.',
+            'Modifies the signature of functions.'
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getPriority()
+    {
+        // must run before ReturnTypeDeclarationFixer and PhpdocNoEmptyReturnFixer
+        return 15;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return PHP_VERSION_ID >= 70100 && $tokens->isTokenKindFound(T_FUNCTION);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isRisky()
+    {
+        return true;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        // These cause syntax errors.
+        static $blacklistFuncNames = [
+            [T_STRING, '__construct'],
+            [T_STRING, '__destruct'],
+            [T_STRING, '__clone'],
+        ];
+
+        for ($index = $tokens->count() - 1; 0 <= $index; --$index) {
+            if (!$tokens[$index]->isGivenKind(T_FUNCTION)) {
+                continue;
+            }
+
+            $funcName = $tokens->getNextMeaningfulToken($index);
+            if ($tokens[$funcName]->equalsAny($blacklistFuncNames, false)) {
+                continue;
+            }
+
+            $startIndex = $tokens->getNextTokenOfKind($index, ['{', ';']);
+
+            if ($this->hasReturnTypeHint($tokens, $startIndex)) {
+                continue;
+            }
+
+            if (!$tokens[$startIndex]->equals('{')) {
+                // No function body defined, fallback to PHPDoc.
+                if ($this->hasVoidReturnAnnotation($tokens, $index)) {
+                    $this->fixFunctionDefinition($tokens, $startIndex);
+                }
+
+                continue;
+            }
+
+            if ($this->hasReturnAnnotation($tokens, $index)) {
+                continue;
+            }
+
+            $endIndex = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_CURLY_BRACE, $startIndex);
+
+            if ($this->hasVoidReturn($tokens, $startIndex, $endIndex)) {
+                $this->fixFunctionDefinition($tokens, $startIndex);
+            }
+        }
+    }
+
+    /**
+     * Determine if there is a non-void return annotation in the function's PHPDoc comment.
+     *
+     * @param Tokens $tokens
+     * @param int    $index  The index of the function token
+     *
+     * @return bool
+     */
+    private function hasReturnAnnotation(Tokens $tokens, $index)
+    {
+        foreach ($this->findReturnAnnotations($tokens, $index) as $return) {
+            if (0 === preg_match(self::VOID_RETURN_PATTERN, $return->getContent())) {
```

Comment:
this will catch false positives here like
```php
/**
 * Is there a @return void annotation ?
 *
 * @return bool
 */
```

Use DocBlock -> getAnnotation('return')